internal/sidecar/service/volumegroup.go

```diff
@@ -0,0 +1,159 @@
+	}, nil
+}
+
+// ModifyVolumeGroupMembership calls CSI-Addons ModifyVolumeGroupMembership service.
+func (vg *VolumeGroupServer) ModifyVolumeGroupMembership(
+	ctx context.Context,
+	req *proto.ModifyVolumeGroupMembershipRequest) (*proto.ModifyVolumeGroupMembershipResponse, error) {
+	// Get the secrets from the k8s cluster
+	data, err := kube.GetSecret(ctx, vg.kubeClient, req.GetSecretName(), req.GetSecretNamespace())
+	if err != nil {
+		klog.Errorf("Failed to get secret %s in namespace %s: %v", req.GetSecretName(), req.GetSecretNamespace(), err)
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	// Get the volumeGroup Id and volumeIds from the request
+	vgReq := &csiVolumeGroup.ModifyVolumeGroupMembershipRequest{
+		VolumeGroupId: req.GetVolumeGroupId(),
+		VolumeIds:     req.GetVolumeIds(),
+		Secrets:       data,
+		Parameters:    req.GetParameters(),
+	}
+
+	_, err = vg.controllerClient.ModifyVolumeGroupMembership(ctx, vgReq)
+	if err != nil {
+		klog.Errorf("Failed to modify volume group: %v", err)
+		return nil, err
+	}
+
+	return &proto.ModifyVolumeGroupMembershipResponse{}, nil
+}
+
+// DeleteVolumeGroup calls CSI-Addons DeleteVolumeGroup service.
+func (vg *VolumeGroupServer) DeleteVolumeGroup(
+	ctx context.Context,
+	req *proto.DeleteVolumeGroupRequest) (*proto.DeleteVolumeGroupResponse, error) {
+	// Get the secrets from the k8s cluster
+	data, err := kube.GetSecret(ctx, vg.kubeClient, req.GetSecretName(), req.GetSecretNamespace())
+	if err != nil {
+		klog.Errorf("Failed to get secret %s in namespace %s: %v", req.GetSecretName(), req.GetSecretNamespace(), err)
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	// Get the volumeGroup Id from the request
+	vgReq := &csiVolumeGroup.DeleteVolumeGroupRequest{
+		VolumeGroupId: req.GetVolumeGroupId(),
+		Secrets:       data,
+	}
+
+	_, err = vg.controllerClient.DeleteVolumeGroup(ctx, vgReq)
+	if err != nil {
+		klog.Errorf("Failed to delete volume group: %v", err)
+		return nil, err
+	}
+
+	return &proto.DeleteVolumeGroupResponse{}, nil
+}
+
+// ControllerGetVolumeGroup calls CSI-Addons ControllerGetVolumeGroup service.
+func (vg *VolumeGroupServer) ControllerGetVolumeGroup(
+	ctx context.Context,
+	req *proto.ControllerGetVolumeGroupRequest) (*proto.ControllerGetVolumeGroupResponse, error) {
+	// Get the secrets from the k8s cluster
+	data, err := kube.GetSecret(ctx, vg.kubeClient, req.GetSecretName(), req.GetSecretNamespace())
+	if err != nil {
+		klog.Errorf("Failed to get secret %s in namespace %s: %v", req.GetSecretName(), req.GetSecretNamespace(), err)
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	// Get the volumeGroup Id from the request
+	vgReq := &csiVolumeGroup.ControllerGetVolumeGroupRequest{
+		VolumeGroupId: req.GetVolumeGroupId(),
+		Secrets:       data,
+	}
+
+	_, err = vg.controllerClient.ControllerGetVolumeGroup(ctx, vgReq)
+	if err != nil {
+		klog.Errorf("Failed to get volume group: %v", err)
+		return nil, err
+	}
```

Comment:
what is the use of this one if we don't fetch the information and surface it to the caller?